### DISCUSSION TITLE
Default `spacing` override on theme for Checkbox?

### DISCUSSION THREAD

**Discussion by eddnav** [chakra-ui/chakra-ui#3265] [Q&A]
I might be missing something here but neither:

```
Checkbox: {
        defaultProps: {
            spacing: "2rem"
        },
    },
```

or

```
Checkbox: {
        baseStyle: {
            spacing: "2rem"
        },
    },
```

Seem to work. Is there a way to set in-theme the default spacing for checkbox?

**Comment by chr-ge** [comment]
@eddnav, when customizing the theme, you can only use [style props](https://chakra-ui.com/docs/features/style-props) and not component props. What you need to do is create a custom checkbox component.

```jsx
import { Checkbox } from "@chakra-ui/react"

const CustomCheckbox = ({ children, ...props }) => (
  <Checkbox spacing={2} {...props}>
   { children }
  </Checkbox>
)

export default CustomCheckbox 
```

**Comment by dodas** [comment]
The `Checkbox`'s `spacing` prop does nothing else but set `marginStart` prop on the label.
You could extend the theme like this:
```jsx
export const theme = extendTheme({
  components: {
    Checkbox: {
      baseStyle: {
        label: {
          marginStart: 10,
        },
      },
    },
  },
})
```

This way tho, the `spacing` prop on `<Checkbox>` won't override the margin set in the theme. That could be fixed in Chakra.
